src/NUnitFramework/tests/Attributes/RangeAttributeTests.cs

```diff
@@ -22,478 +22,489 @@
 // ***********************************************************************
 
 using System;
-using System.Reflection;
+using System.Collections.Generic;
+using System.Linq;
 using NUnit.Framework.Internal;
+using NUnit.TestData;
 using NUnit.TestUtilities;
 
-#if NETCOREAPP1_1
-using System.Linq;
-#endif
-
 namespace NUnit.Framework.Attributes
 {
-    public class RangeAttributeTests
+    public partial class RangeAttributeTests
     {
-        #region Ints
-
-        [Test]
-        public void IntRange()
-        {
-            CheckValues(nameof(MethodWithIntRange), 11, 12, 13, 14, 15);
-        }
+        #region Shared specs
+
+        public static IEnumerable<Type> TestedParameterTypes() => new[]
+        {
+            typeof(sbyte),
+            typeof(byte),
+            typeof(short),
+            typeof(ushort),
+            typeof(int),
+            typeof(uint),
+            typeof(long),
+            typeof(ulong),
+            typeof(float),
+            typeof(double),
+            typeof(decimal)
+        };
+
+        private static readonly Type[] Int32RangeConvertibleToParameterTypes = { typeof(int), typeof(sbyte), typeof(byte), typeof(short), typeof(decimal) };
+        private static readonly Type[] UInt32RangeConvertibleToParameterTypes = { typeof(uint) };
+        private static readonly Type[] Int64RangeConvertibleToParameterTypes = { typeof(long) };
+        private static readonly Type[] UInt64RangeConvertibleToParameterTypes = { typeof(ulong) };
+        private static readonly Type[] SingleRangeConvertibleToParameterTypes = { typeof(float) };
+        private static readonly Type[] DoubleRangeConvertibleToParameterTypes = { typeof(double), typeof(decimal) };
```

Comment:
Shouldn't these start with a lowercase? (also it took me some time to find the connection between these and `ParamAttributeTypeConversions.TryConvert` 😄 .

Reply:
I'll add a comment pointing to the XML docs I'm adding on ParamAttributeTypeConversions. 📄

Reply:
Looping back to say that the outcome of https://github.com/nunit/nunit/issues/2822 was to allow this naming convention.